<launch>

<arg name="mav_name" default="ibis" />
<arg name="namespace" default="$(arg mav_name)" />
<arg name="process_every_nth_frame" default="1" />
<arg name="stereo_params_camchain" default="$(find mav_startup)/parameters/mavs/ibis/camchain_p22031.yaml"/>

<group ns="$(arg namespace)">
 
    <node name="dense_stereo" pkg="image_undistort" type="dense_stereo_node">
      <param name="input_camera_info_from_ros_params" value = "true"/>
      <param name="first_camera_namespace" value="cam0"/>
      <param name="second_camera_namespace" value="cam1"/>
      <param name="first_output_frame" value="cam0_rect"/>
      <param name="second_output_frame" value="cam1_rect"/>
      <param name="scale" value="1.0"/>
      <param name="process_every_nth_frame" value="$(arg process_every_nth_frame)"/>
      <param name="depth/use_sgbm" value="true"/>
      <param name="depth/do_median_blur" value="false"/>
      <param name="depth/use_mode_HH" value="true"/>
      <rosparam file="$(arg stereo_params_camchain)"/>
  
      <remap from="raw/first/image" to="/$(arg mav_name)/cam0/image_raw"/>
      <remap from="raw/second/image" to="/$(arg mav_name)/cam1/image_raw"/>
      <remap from="raw/first/camera_info" to="/$(arg mav_name)/cam0/camera_info"/>
      <remap from="raw/second/camera_info" to="/$(arg mav_name)/cam1/camera_info"/>
    </node>
</group>

</launch>